gravity.sh

```diff
@@ -520,11 +520,18 @@ gravity_DownloadBlocklists() {
 }
 
 parseList() {
-  local adlistID="${1}" src="${2}" target="${3}" temp_file non_domains sample_non_domains
+  local adlistID="${1}" src="${2}" target="${3}" temp_file temp_file_descriptor non_domains sample_non_domains
 
   # Create a temporary file for the sed magic instead of using "${target}" directly
   # this allows to split the sed commands to improve readability
-  temp_file="$(mktemp -p "/tmp" --suffix=".gravity")"
+  # we use a file descriptor here and remove the temporary file immediately so the content will be delated in any case
+  # when the script stops
+    temp_file="$(mktemp -p "/tmp" --suffix=".gravity")"
+    temp_file_descriptor=$(ReturnFileDescriptor)
+    eval "exec $temp_file_descriptor>$temp_file"
+    rm "${temp_file}"
+    temp_file="/proc/$$/fd/$temp_file_descriptor"
+
 
```

Comment:
~~If you are going to break out getting a file descriptor to a separate function then go ahead and split out creating the temporary file handle to a function as well.~~

~~Or if you don't feel that making the temp file is something that needs its own function then there's no need to have the file descriptor as a function either.~~

On second thought, I really don't like making `bash` return values from functions using tricks like `echo`. It's way too easy for that to break. Don't split out things to functions that don't set a global variable. In this case keep everything in the same function, there's no need for the added complexity. If there's a need in the future to factor out the functions then we can address that then.